echidna/core/spectra.py

```diff
@@ -1,4 +1,226 @@
+import collections
 import numpy
+import yaml
+import copy
+
+
+class SpectraParameter(object):
+    """Simple data container that holds information for a Spectra parameter
+    (i.e. axis of the spectrum).
+
+    Args:
+      name (str): The name of this parameter
+      low (float): The lower limit of this parameter
+      high (float): The upper limit of this parameter
+      bins (int): The number of bins for this parameter
+
+    Attributes:
+      _name (str): The name of this parameter
+      _low (float): The lower limit of this parameter
+      _high (float): The upper limit of this parameter
+      _bins (int): The number of bins for this parameter
+      _dimension (int): The position in the data array for this parameter.
+    """
+
+    def __init__(self, name, low, high, bins):
+        """Initialise SpectraParameter class
+        """
+        self._name = name
+        self._high = high
+        self._low = low
+        self._bins = bins
+
+    def set_par(self, **kwargs):
+        """Set a limit / binning parameter after initialisation.
+
+        Args:
+          kwargs (dict): keyword arguments
+
+        .. note::
+
+          Keyword arguments include:
+
+            * low (float): Value to set the lower limit to of the parameter
+            * high (float): Value to set the higher limit to of the parameter
+            * bins (int): Value to set the number of bins of the parameter
+
+        Raises:
+          TypeError: Unknown variable type passed as a kwarg.
+        """
+        for kw in kwargs:
+            if kw == "low":
+                self._low = float(kwargs[kw])
+            elif kw == "high":
+                self._high = float(kwargs[kw])
+            elif kw == "bins":
+                self._bins = int(kwargs[kw])
+            else:
+                raise TypeError("Unhandled parameter name / type %s" % kw)
+
+    def get_width(self):
+        """Get the width of the binning for the parameter
+
+        Returns:
+          float: Bin width.
+        """
+        return (self._high - self._low) / float(self._bins)
+
+    def get_unit(self):
+        """Get the default unit for a given parameter
+
+        Raises:
+          Exception: Unknown parameter.
+
+        Returns:
+          string: Unit of the parameter
+        """
+        if self._name.split['_'][0] == "energy":
```

Comment:
Assuming the`par.get_unit()` change, I then get an error on this line. Should have normal (not square)brackets around split delimiter `('_')`